JcloudsLocation is not releasing its customizers (BROOKLYN-427)

geomacy opened the pull request:
preRelease and postRelease of JcloudsLocationCustomizers are not being 
called as expected during the release of a Jclouds machine location.
The problem is that the "setup" used in JcloudsLocation.obtainOnce includes a copy of the flags passed from MachineLifecycleEffectorTasks#stopAnyProvisionedMachines,
which includes the customizers. However, the "setup" used in the "release"
method of JcloudsLocation is taken from the JcloudsLocation itself, which
is the provisioning (i.e. parent) location, which doesn't have those 
customizers in it.
It would actually make more sense to get the location customizers from
the machine location, as a given parent could have mutiple machines
with different customizers.

The changes here include an addition to the
`JcloudsCustomizerInstantiationYamlDslTest#testCustomizers`
to demonstrate the problem, and a fix that adds the config to 
the machine location.

neykov commented on locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/JcloudsLocation.java:
```diff
@@ -2219,7 +2227,7 @@ public void release(MachineLocation rawMachine) {
         Exception tothrow = null;
 
-        ConfigBag setup = config().getBag();
+        ConfigBag setup = ((LocationInternal)machine).config().getBag();
```
Is the cast needed?

geomacy replied:
IIRC it is needed to be able to do the `getBag()`

neykov commented on locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/JcloudsLocation.java:
```diff
@@ -1065,6 +1066,13 @@ protected MachineLocation obtainOnce(ConfigBag setup) throws NoMachinesAvailable
+            }
+            if (machineCustomizers.size() > 0) {
+                machineLocation.config().set(MACHINE_LOCATION_CUSTOMIZERS, machineCustomizers);
+            }
```
Can you instead add the config for `JCLOUDS_LOCATION_CUSTOMIZERS` at [construction time](https://github.com/apache/brooklyn-server/blob/master/locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/JcloudsLocation.java#L2056-L2099) so migrated settings are all at one place.

geomacy commented:
Addressed review [comment](https://github.com/apache/brooklyn-server/pull/530#discussion_r97329773) "Can you instead add the config for JCLOUDS_LOCATION_CUSTOMIZERS..."

geomacy commented:
test failure unrelated?

geomacy commented:
let's try again

neykov commented:
Thanks for updates, merging.
